### DISCUSSION TITLE
Global Providers like NestJS

### DISCUSSION THREAD

**Discussion by LukasGaebler** [fastapi/fastapi#10475] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
As this is just a question, I dont really have code...
```

### Description

Is it possible to use providers in fastapi just like in nestjs? I tried to figure it out with dependency injection but i was unable to. I want to inject a class and then call a special method from that class. 

### Operating System

macOS

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

3.11.5

**Comment by pythonweb2** [comment]
Injecting a class as a dependency is explained here: https://fastapi.tiangolo.com/advanced/advanced-dependencies/#a-callable-instance
